File: algorithm/leetcode209_minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int s, vector<int>& nums) {
    	if(nums.size() == 0)
    		return 0;
        int i = 0, j = 0, len = INT_MAX, sum = nums[0];
        while(j < nums.size())
        {
        	if(sum < s)
        	{
        		j++;
        		if(j < nums.size())
        		sum += nums[j];
        	}
        	else if(sum >= s)
        	{
        		if(j-i+1 < len)
        			len = j-i+1;
        		sum -= nums[i];
        		i++;
        	}
        }
        if(len == INT_MAX)
        	return 0;
        return len;
    }
};

File: algorithm/Python3_version/leetcode209.minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, s: int, nums: List[int]) -> int:
        minlen = float('inf')
        l, r, ans = 0, 0, 0
        for r in range(len(nums)):
            ans += nums[r]
            while ans >= s:
                minlen = min(minlen, r-l+1)
                ans -= nums[l]
                l += 1
        return minlen if minlen!=float('inf') else 0